src/Core/Houston.vala

```diff
@@ -42,29 +42,30 @@ public class AppCenterCore.Houston : Object {
         return root;
     }
 
-    public string[] get_newest () {
+    public async string[] get_newest () {
         var uri = HOUSTON_API_URL + "/newest";
+        string[] app_ids = {};
 
         debug ("Requesting newest applications from %s", uri);
 
         var message = new Soup.Message ("GET", uri);
-        session.send_message (message);
-
-        string[] app_ids = {};
-        try {
-            var res = process_response ((string) message.response_body.data);
-
-            if (res.has_member ("data")) {
-                var data = res.get_array_member ("data");
-
-                foreach (var id in data.get_elements ()) {
-                    app_ids += ((string) id.get_value ());
+        session.queue_message (message, (sess, mess) => {
+            try {
+                var res = process_response ((string) mess.response_body.data);
+                if (res.has_member ("data")) {
+                    var data = res.get_array_member ("data");
+
+                    foreach (var id in data.get_elements ()) {
+                        app_ids += ((string) id.get_value ());
+                    }
                 }
+            } catch (Error e) {
+                stderr.printf ("Houston: %s\n", e.message);
             }
-        } catch (Error e) {
-            stderr.printf ("Houston: %s\n", e.message);
-        }
+            Idle.add (get_newest.callback);
+        });
 
+        yield;
```

Comment:
Why do you yield here right before return? I can't see any other yield statements in this method and only one yield statement right before return seems to be rather missing the point.